libcloudforensics/providers/aws/internal/ebs.py

```diff
@@ -119,22 +120,24 @@ def Snapshot(self, snapshot_name: Optional[str] = None) -> 'AWSSnapshot':
       RuntimeError: If the snapshot could not be created.
     """
 
+    if not tags:
+      tags = {}
+
+    snapshot_name = tags.get('Name') or self.volume_id
     timestamp = datetime.now().strftime('%Y%m%d%H%M%S')
-    if not snapshot_name:
-      snapshot_name = self.volume_id
     truncate_at = 255 - len(timestamp) - 1
     snapshot_name = '{0}-{1}'.format(snapshot_name[:truncate_at], timestamp)
```

Comment:
Is it desired here to add a timestamp suffix to the name in case the name is given by the user as a parameter? or should it only be in case no name is given? I am thinking in a case where the user create a snapshot of the volume with a certain name tag and then search for the same name tag later in the code and find no match, but I am not fully aware how tags work in AWS. If this makes sense to you:

```suggestion
      truncate_at = 255 - len(timestamp) - 1
      snapshot_name = '{0}-{1}'.format(snapshot_name[:truncate_at], timestamp)
```

Reply:
I think this was initially to be on par with what we had in GCP; however there it is required since the snapshot name is used as ID by GCP (the timestamp was there to limit name collisions). This doesn't matter in AWS since tags are not unique between resources anyways.

I removed it, and opted to go with `volume_id + '-snapshot'` for the default snapshot name if no tags are provided.